src/connection/adapter/base_http_adapter.ts

```diff
@@ -294,17 +294,30 @@ export abstract class BaseHttpAdapter implements Connection {
   }
 
   private logResponse(
+    request: Http.ClientRequest,
     params: RequestParams,
     response: Http.IncomingMessage,
     startTimestamp: number
   ) {
+    const getRequestHeaders = () => {
```

Comment:
why not inline values instead of using creating functions on every call? It adds GC pressure

Reply:
Ouch, that's the case indeed. I thought it will be at least hoisted. Turned out it is not correct for this definition.